tutorial-1-translating-with-jobs/java/src/main/java/com/smartling/samples/jobstutorial/ProcessJob.java

```diff
@@ -0,0 +1,198 @@
+import com.smartling.api.jobs.v3.TranslationJobsApi;
+import com.smartling.api.jobs.v3.TranslationJobsApiFactory;
+import com.smartling.api.jobs.v3.pto.ContentProgressReportPTO;
+import com.smartling.api.jobs.v3.pto.TranslationJobCreateCommandPTO;
+import com.smartling.api.jobs.v3.pto.TranslationJobCreateResponsePTO;
+import com.smartling.api.v2.authentication.AuthenticationApi;
+import com.smartling.api.v2.authentication.AuthenticationApiFactory;
+import com.smartling.api.v2.client.ClientFactory;
+import com.smartling.api.v2.client.DefaultClientConfiguration;
+import com.smartling.api.v2.client.auth.Authenticator;
+import com.smartling.api.v2.client.auth.BearerAuthSecretFilter;
+import com.smartling.api.v2.client.exception.RestApiRuntimeException;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+
+public class ProcessJob
+{
+    private static final List<String> LOCALES = new ArrayList<>(Arrays.asList("fr-FR"));
+    private static final List<String> FILE_LIST = new ArrayList<>(Arrays.asList("site-navigation.json","products.json"));
+    private static final String FILE_TYPE = "json";
+
+    public static void main( String[] args ) throws Exception
+    {
+        // Read command-line arguments
+        if (args.length != 1) {
+            System.out.println("Usage: mvn exec:java@run");
+            System.out.println("(See pom.xml to configure parameters.)");
+            System.exit(1);
+        }
+        String jobName = args[0];
+
+        // Read authentication credentials from environment
+        String userId = System.getenv("DEV_USER_IDENTIFIER");
+        String userSecret = System.getenv("DEV_USER_SECRET");
+        String projectId = System.getenv("DEV_PROJECT_ID");
+        if (userId == null || userSecret == null || projectId == null) {
+            System.out.println("Missing environment variables. Did you run setenv?");
+            System.exit(1);
+        }
+
+        // Create factory for building API clients
+        DefaultClientConfiguration clientConfiguration = DefaultClientConfiguration.builder().build();
+        ClientFactory clientFactory = new ClientFactory();
+
+        // Create the authenticator to be used by subsequent API calls
+        AuthenticationApi authenticationApi = new AuthenticationApiFactory(clientFactory).buildApi(clientConfiguration);
+        Authenticator authenticator = new Authenticator(userId, userSecret, authenticationApi);
+
+        try {
+            System.out.println();
+
+            // create job
+            System.out.println("Calling 'create job' endpoint...");
+            TranslationJobsApi jobsApi = new TranslationJobsApiFactory(clientFactory)
+                    .buildApi(new BearerAuthSecretFilter(authenticator), clientConfiguration);
```

Comment:
You can create `BearerAuthSecretFilter` right after `Authenticator authenticator = new..` line and reuse it too.

Reply:
Done.